internal/types/array_test.go

```diff
@@ -77,3 +77,78 @@ func TestArray(t *testing.T) {
 		assert.Equal(t, int32(42), b.s[0])
 	})
 }
+
+func TestArrayContains(t *testing.T) {
+	for name, tc := range map[string]struct {
+		array    *Array
+		filter   any
+		expected bool
+	}{
+		"String": {
+			array:    must.NotFail(NewArray("foo", "bar")),
+			filter:   "foo",
+			expected: true,
+		},
+		"StringNegative": {
+			array:    must.NotFail(NewArray("foo", "bar")),
+			filter:   "hello",
+			expected: false,
+		},
+		"Int32": {
+			array:    must.NotFail(NewArray(int32(42), int32(43), int32(45))),
+			filter:   int32(43),
+			expected: true,
+		},
+		"Int32Negative": {
+			array:    must.NotFail(NewArray(int32(42), int32(43), int32(45))),
+			filter:   int32(44),
+			expected: false,
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			assert.Equal(t, tc.expected, tc.array.Contains(tc.filter))
```

Comment:
We typically use `testutil.AssertEqual` that handles NaNs

Reply:
Fixed. I'd say the other function in this file - `TestArray` needs to be refactored to follow the same standard. Also, I'd separate tests for different `Array` ops, then one doesn't need to run tests for everything if they want to test or play with a particular operation... But it's not in scope if this task :)